MapboxNavigation/NavigationMapView.swift

```diff
@@ -1035,6 +1043,30 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         return MGLShapeCollectionFeature(shapes: [mainRoute])
     }
     
+    /**
+     Returns an array of `MapboxStreetsRoadClass` objects for specific leg.
+     
+     - parameter leg: Leg of the route.
+     - returns: An array of `MapboxStreetsRoadClass` for specific `RouteLeg`. `MapboxStreetsRoadClass` will be set to `nil` if it's not present in `Intersection`.
+     */
+    func streetsRoadClasses(_ leg: RouteLeg) -> [MapboxStreetsRoadClass?] {
+        // Pick only valid segment indices for specific `Intersection` in `RouteStep`.
+        // Array of segment indexes can look like this: [0, 3, 24, 28, 48, 50, 51, 53].
+        let segmentIndices = leg.steps.compactMap({ $0.segmentIndicesByIntersection?.compactMap({ $0 }) }).reduce([], +)
```

Comment:
I think `RouteLeg.segmentRangesByStep` may simplify this computation.

Reply:
Seems that `segmentRangesByStep` is not detailed enough, as it stores only lower and upper bounds. `segmentIndicesByIntersection` is more precise in this case as it stores segment indices for `Intersection`.